src/cow-react/modules/limitOrders/hooks/useApplyLimitRate.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback } from 'react'
+import { useAtom } from 'jotai'
+
+import { Field } from 'state/swap/actions'
+import { limitRateAtom } from '../state/limitRateAtom'
+
+// Rounds number only if the number of decimals goes above the decimals param
+export const limitDecimals = (amount: number, decimals: number): number => {
```

Comment:
Yea I agree with all of this :smile: 
I suggest for now and the easier implementation we use simple values like regular numbers and strings because of the complexity. And once we are done with basic implementation we switch to better value types.